sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpEventHubClientTests.cs

```diff
@@ -0,0 +1,322 @@
+                .Returns(Task.FromResult(new AccessToken(tokenValue, DateTimeOffset.MaxValue)))
+                .Verifiable();
+
+            mockConverter
+                .Setup(converter => converter.CreateEventHubPropertiesRequest(It.Is<string>(value => value == eventHubName), It.Is<string>(value => value == tokenValue)))
+                .Returns(default(AmqpMessage))
+                .Callback(() => cancellationSource.Cancel())
+                .Verifiable();
+
+            // Because the AMQP library is not friendly to mocking, only the path up to conversion can be tested without external
+            // dependencies.  To ensure that execution stops after that point, the converter triggers cancellation that should take
+            // place immediately following the conversion and result in a well-known exception.
+
+            var client = new InjectableMockClient("my.eventhub.com", eventHubName, mockCredential.Object, new EventHubClientOptions(), Mock.Of<EventHubRetryPolicy>(), null, mockConverter.Object);
+            Assert.That(async () => await client.GetPropertiesAsync(cancellationSource.Token), Throws.InstanceOf<TaskCanceledException>());
+
+            mockCredential.VerifyAll();
+            mockConverter.VerifyAll();
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpEventHubClient.GetPartitionPropertiesAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void GetPartitionPropertiesAsyncValidatesThePartition(string partition)
+        {
+           var client = new AmqpEventHubClient("my.eventhub.com", "somePath", Mock.Of<TokenCredential>(), new EventHubClientOptions(),  Mock.Of<EventHubRetryPolicy>());
+           Assert.That(async () => await client.GetPartitionPropertiesAsync(partition, CancellationToken.None), Throws.ArgumentException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpEventHubClient.GetPartitionPropertiesAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void GetPartitionPropertiesAsyncRespectsTheCancellationTokenIfSetWhenCalled()
+        {
+           var cancellationSource = new CancellationTokenSource();
+           cancellationSource.Cancel();
+
+           var client = new AmqpEventHubClient("my.eventhub.com", "somePath", Mock.Of<TokenCredential>(), new EventHubClientOptions(),  Mock.Of<EventHubRetryPolicy>());
+           Assert.That(async () => await client.GetPartitionPropertiesAsync("Fred", cancellationSource.Token), Throws.InstanceOf<TaskCanceledException>());
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpEventHubClient.GetPartitionPropertiesAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void GetPartitionPropertiesAsyncCreatesTheRequest()
```

Comment:
Due to limitations in the AMQP library, this was as deep as we could test without providing another layer of abstraction around connections; since this is already a type dedicated to AMQP, that seemed excessive.   Full communication will be verified by Live test.